pyresample/test/test_geometry/test_area.py

```diff
@@ -0,0 +1,2003 @@
+        adef = area_class.from_cf(ds)
+
+        assert adef.shape == (19, 37)
+        xc = adef.projection_x_coords
+        yc = adef.projection_y_coords
+        assert xc[0] == -180., "Wrong x axis (index 0)"
+        assert xc[1] == -180. + 10.0, "Wrong x axis (index 1)"
+        assert yc[0] == -90., "Wrong y axis (index 0)"
+        assert yc[1] == -90. + 10.0, "Wrong y axis (index 1)"
+
+    def test_area_def_init_projection(self, create_test_area):
+        """Test AreaDefinition with different projection definitions."""
+        proj_dict = {
+            'a': '6378144.0',
+            'b': '6356759.0',
+            'lat_0': '90.00',
+            'lat_ts': '50.00',
+            'lon_0': '8.00',
+            'proj': 'stere'
+        }
+        crs = CRS(CRS.from_dict(proj_dict).to_wkt())
+        # pass CRS object directly
+        area_def = create_test_area(
+            crs,
+            800,
+            800,
+            (-1370912.72, -909968.64000000001, 1029087.28, 1490031.3600000001))
+        assert crs == area_def.crs
+        # PROJ dictionary
+        area_def = create_test_area(
+            crs.to_dict(),
+            800,
+            800,
+            (-1370912.72, -909968.64000000001, 1029087.28, 1490031.3600000001))
+        assert crs == area_def.crs
+        # PROJ string
+        area_def = create_test_area(
+            crs.to_string(),
+            800,
+            800,
+            (-1370912.72, -909968.64000000001, 1029087.28, 1490031.3600000001))
+        assert crs == area_def.crs
+        # WKT2
+        area_def = create_test_area(
+            crs.to_wkt(),
+            800,
+            800,
+            (-1370912.72, -909968.64000000001, 1029087.28, 1490031.3600000001))
+        assert crs == area_def.crs
+        # WKT1_ESRI
+        area_def = create_test_area(
+            crs.to_wkt(version='WKT1_ESRI'),
+            800,
+            800,
+            (-1370912.72, -909968.64000000001, 1029087.28, 1490031.3600000001))
+        # WKT1 to WKT2 has some different naming of things so this fails
+        # assert crs == area_def.crs
+
+    def test_areadef_immutable(self, create_test_area):
+        """Test that some properties of an area definition are immutable."""
+        area_def = create_test_area(
+            {
+                'a': '6378144.0',
+                'b': '6356759.0',
+                'lat_0': '50.00',
+                'lat_ts': '50.00',
+                'lon_0': '8.00',
+                'proj': 'stere'
+            },
+            10,
+            10,
+            (-1370912.72, -909968.64000000001, 1029087.28, 1490031.3600000001))
+        with pytest.raises(AttributeError):
+            area_def.shape = (10, 10)
+        with pytest.raises(AttributeError):
+            area_def.proj_str = "seaweed"
+        with pytest.raises(AttributeError):
+            area_def.area_extent = (-1000000, -900000, 1000000, 1500000)
+
+
+class TestAreaDefinitionMetadata:
```

Comment:
I'm not sure about the name here. At least, should this class and all that is future-specific be moved to a future test file?

Reply:
I thought about this a lot when I started this PR and I think I still agree with past Dave. Or at least I'm 50/50 on it so I can't say I feel strongly. The original plan (and how it is being implemented now) is that this module will test legacy and future implementations and the legacy test module will eventually be removed. So I didn't want to confuse contributions for the legacy area definition by telling people "I know you want to put the tests in `test/`, but they should actually go in `test/future/` even though that's not where your changes went.